### PR TITLE
feat: billing

### PR DESCRIPTION
**Pull request by ArmanNik**
## What does this PR do?

Billing pages and logic

(Provide a description of what this PR does.)

## Test Plan

Manual

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: src/routes/console/changeOrganizationTierCloud.svelte [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            await invalidate(Dependencies.ACCOUNT);
+            dispatch('created');
+            await goto(`/console/organization-${org.$id}`);
+            if ($isUpgrade) {
+                addNotification({
+                    type: 'success',
+                    isHtml: true,
+                    message: `
+                    <b>Your organization has been upgraded</b>
+                   make the most of your increased resource capacity and continue building great things with Appwrite.`
```
Comment (gewenyu99) [inline src/routes/console/changeOrganizationTierCloud.svelte:69]: Can we keep it to just "Your organization has been upgraded" ?

@heyCarla @ArmanNik This is too much to read for how long this stays on screen, and a bit tacky

**Review by gewenyu99 — DISMISSED** [review]

File: src/routes/console/changeOrganizationTierCloud.svelte (hunk 1 above)
Comment (heyCarla) [reply src/routes/console/changeOrganizationTierCloud.svelte:69]: Fine by me.

File: src/routes/console/organization-[organization]/billing/planSummary.svelte [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            : $organization?.billingPlan === 'tier-1'
+            ? tierPro
+            : tierFree;
+
+    $: aggregation = $aggregationList.aggregations[0];
+</script>
+
+{#if $organization}
+    <CardGrid>
+        <Heading tag="h2" size="6">Plan summary</Heading>
```
Comment (gewenyu99) [inline src/routes/console/organization-[organization]/billing/planSummary.svelte:25]:
```suggestion
        <Heading tag="h2" size="6">Plan</Heading>
```

Comment (heyCarla) [reply src/routes/console/organization-[organization]/billing/planSummary.svelte:25]: As discussed we'll keep this as "Plan summary" for now.

File: src/routes/console/wizard/cloudOrganization/confirmDetails.svelte [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    async function fetchCard() {
+        try {
+            const card = await sdk.forConsole.billing.getPaymentMethod(
+                $createOrganization.paymentMethodId
+            );
+            console.log(card);
+            return card;
+        } catch (error) {
+            console.log(error);
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganization/confirmDetails.svelte:25]: leftovers

File: src/routes/console/wizard/cloudOrganization/inviteMembers.svelte [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    function removeCollaborator(email: string) {
+        $createOrganization.collaborators = $createOrganization.collaborators.filter(
+            (collaborator) => collaborator !== email
+        );
+    }
+
+    const plan = $plansInfo.plans.find((p) => p.$id === $createOrganization.billingPlan);
+
+    $: console.log(plan);
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganization/inviteMembers.svelte:35]: leftover

File: src/routes/console/wizard/cloudOrganizationChangeTier/choosePlan.svelte [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+    onMount(async () => {
+        usage = await sdk.forConsole.billing.listUsage($organization.$id);
+        members = await sdk.forConsole.teams.listMemberships($organization.$id);
+
+        //Select closest tier from starting one
+        // if ($organization.billingPlan === 'tier-2') {
+        //     $changeOrganizationTier.billingPlan = 'tier-1';
+        // }
+        // else
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganizationChangeTier/choosePlan.svelte:82]: leftover?

File: src/routes/console/wizard/cloudOrganizationChangeTier/confirmDetails.svelte [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+    let comment: string = null;
+
+    async function fetchCard() {
+        try {
+            const card = await sdk.forConsole.billing.getPaymentMethod(
+                $changeOrganizationTier.paymentMethodId
+            );
+            return card;
+        } catch (error) {
+            console.log(error);
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganizationChangeTier/confirmDetails.svelte:39]: leftover

File: src/routes/console/wizard/cloudOrganizationChangeTier/paymentDetails.svelte [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            const paymentMethod = symmetricDifference(
+                methods.paymentMethods,
+                latestMethods.paymentMethods
+            )[0] as PaymentMethodData;
+
+            console.log(paymentMethod);
+            $changeOrganizationTier.paymentMethodId = paymentMethod.$id;
+            invalidate(Dependencies.PAYMENT_METHODS);
+        } catch (e) {
+            console.log(e.message);
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganizationChangeTier/paymentDetails.svelte:52]: leftover

File: src/routes/console/wizard/cloudOrganizationChangeTier/paymentDetails.svelte [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+        try {
+            await submitStripeCard(name);
+            const latestMethods = await sdk.forConsole.billing.listPaymentMethods();
+            const paymentMethod = symmetricDifference(
+                methods.paymentMethods,
+                latestMethods.paymentMethods
+            )[0] as PaymentMethodData;
+
+            console.log(paymentMethod);
```
Comment (TorstenDittmann) [inline src/routes/console/wizard/cloudOrganizationChangeTier/paymentDetails.svelte:48]: leftover

File: src/lib/components/fakeModal.svelte [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    export let size: 'small' | 'big' = 'big';
+    export let icon: string = null;
+    export let state: 'success' | 'warning' | 'error' | 'info' = null;
+    export let error: string = null;
+    export let closable = true;
+    export let headerDivider = true;
+    export let onSubmit: (e: SubmitEvent) => Promise<void> | void = function () {
+        return;
+    };
+    export let title = '';
```
[the remark this answers is not in the document]
Comment (ArmanNik) [reply src/lib/components/fakeModal.svelte:18]: Yes we allow titles to be passed by slot for complex implemrntations

File: src/lib/layout/unauthenticated.svelte [hunk 10]
```diff
@@ -5,32 +5,100 @@
...
+        </div>
+    </section>
+</main>
+
+<!-- OLD one -->
+<!--
+<main class="grid-1-1 is-full-page" id="main">
+    <section
+            class="u-flex u-flex-vertical"
+            style:--url={`url(${$app.themeInUse === 'dark' ? imgDark : imgLight})`}>
```
[the remark this answers is not in the document]
Comment (ArmanNik) [reply src/lib/layout/unauthenticated.svelte:85]: After the launch we are probably going to revert to this so it's worth keeping for now

File: src/routes/console/wizard/cloudOrganizationChangeTier/choosePlan.svelte (hunk 5 above)
Comment (ArmanNik) [reply src/routes/console/wizard/cloudOrganizationChangeTier/choosePlan.svelte:84]: We need this code when Scale plan is released

File: src/routes/console/organization-[organization]/billing/availableCredit.svelte [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+    let creditList: CreditList = {
+        credits: [],
+        total: 0
+    };
+    let show = false;
+    let reloadOnWizardClose = false;
+
+    onMount(async () => {
+        request();
+    });
```
[the remark this answers is not in the document]
Comment (TGlide) [reply src/routes/console/organization-[organization]/billing/availableCredit.svelte:32]: ^

**Review by TGlide — APPROVED** [review]

**Review by eldadfux — APPROVED** [review]

**Pull request merged**